spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -268,56 +266,37 @@ protected MessageGroup copy(MessageGroup group) {
 	}
 
 	@Override
-	public void addMessagesToGroup(Object groupId, Message<?>... messages) {
```

Comment:
I think it is OK to revert the previous version of the method and just don't use `doAddMessagesToGroup()` contract.
This way that custom logic for the lock and `UpperBound` would be met and test will work again.

Reply:
The code revert is ok, but this leads to the question of whether `doAddMessagesToGroup` is a common design

Reply:
The code revert is ok, but this leads to the question of whether `doAddMessagesToGroup` is a common design?